src/main/java/at/hannibal2/skyhanni/features/inventory/ItemPickupLog.kt

```diff
@@ -0,0 +1,314 @@
+    private var itemsRemovedFromInventory = mutableMapOf<Int, UpdatedItem>()
+    private var display: Renderable = Renderable.string("")
+
+    private val patternGroup = RepoPattern.group("itempickuplog")
+    private val shopPattern by patternGroup.pattern(
+        "shoppattern",
+        "^(?<itemName>.+?)(?: x\\d+)?\$",
+    )
+
+    private val bannedItemsPattern by patternGroup.list(
+        "banneditems",
+        "SKYBLOCK_MENU",
+        "CANCEL_PARKOUR_ITEM",
+        "CANCEL_RACE_ITEM",
+    )
+    private val bannedItemsConverted = bannedItemsPattern.map { it.toString().asInternalName() }
+
+
+    @SubscribeEvent
+    fun onRenderOverlay(event: GuiRenderEvent) {
+        if (!isEnabled()) return
+        config.pos.renderRenderable(display, posLabel = "Item Pickup Log Display")
+    }
+
+    @SubscribeEvent
+    fun onWorldChange(event: LorenzWorldChangeEvent) {
+        if (!isEnabled()) return
+        itemList.clear()
+        itemsAddedToInventory.clear()
+        itemsRemovedFromInventory.clear()
+    }
+
+    @SubscribeEvent
+    fun onSackChange(event: SackChangeEvent) {
+        if (!isEnabled() || !config.sack) return
+
+        event.sackChanges.forEach {
+            val itemStack = (it.internalName.getItemStack())
+            val item = UpdatedItem(itemStack.displayName, it.delta.absoluteValue.toLong(), it.internalName)
+
+            updateItem(itemStack.hash(), item, itemStack, it.delta < 0)
+        }
+    }
+
+    @SubscribeEvent
+    fun onPurseChange(event: PurseChangeEvent) {
+        if (!isEnabled() || !config.coins || !worldChangeCooldown()) return
+
+        updateItem(0, UpdatedItem("§6Coins", event.coins.absoluteValue.toLong(), coinIcon), coinIcon.getItemStack(), event.coins < 0)
+    }
+
+    @SubscribeEvent
+    fun onTick(event: LorenzTickEvent) {
+        if (!isEnabled()) return
+
+        val oldItemList = mutableMapOf<Int, Pair<ItemStack, Int>>()
+
+        itemList.forEach {
+            oldItemList[it.key] = it.value
+        }
+
+        itemList.clear()
+
+        val inventoryItems = InventoryUtils.getItemsInOwnInventory().toMutableList()
+        val cursorItem = Minecraft.getMinecraft().thePlayer.inventory?.itemStack
+
+        if (cursorItem != null) {
+            val hash = cursorItem.hash()
+            //this prevents items inside hypixel guis counting when picked up
+            if (oldItemList.contains(hash)) {
+                inventoryItems.add(cursorItem)
+            }
+        }
+
+        for (itemStack in inventoryItems) {
+            val hash = itemStack.hash()
+            val old = itemList[hash]
+            if (old != null) {
+                itemList[hash] = old.copy(second = old.second + itemStack.stackSize)
+            } else {
+                itemList[hash] = itemStack to itemStack.stackSize
+            }
+        }
+
+        if (!worldChangeCooldown()) return
+
+        oldItemList.forEach {
```

Comment:
this can be merged with itemList below (remove code duplication)

Reply:
i don't really see how this could be merged as they seem to do diffferent things, I am open to discussion if i've missed something <3

Reply:
those two code parts do the same, except on different maps, and with one value being different.
![image](https://github.com/user-attachments/assets/580b2e6a-8e02-41c2-875c-f9995b2aadfc)